CONTRIBUTING.rst

```diff
@@ -2,69 +2,77 @@
+Very large pull requests may receive only a partial review at first.
+This is to be able to respond within a reasonably short time frame while
+still taking time to fairly assess all the changes.
 
-1. Shebang line, #! /usr/bin/env python (only for executable scripts)
-2. License header (``LICENSE_HEADER.txt``) and module-level comments
-3. Module-level docstring
-4. ``__all__ = [...]`` statement, if present
-5. Imports (alphabetically ordered within each block)
+We use CODING_GUIDELINES_ as a reference for reviewing contributions from
+a code style and design standpoint.
 
-   a. Block of imports from the standard library
-   b. Block of imports from general third party libraries (e.g. numpy,
-      xarray)
-   c. Block of imports from specific submodules of the project
+Further points to be considered are:
 
-6. Private module variables, functions and classes (names start with
-   underscore)
-7. Public module variables, functions and classes
+- Check all authors are covered by a contributor agreement and added to AUTHORS_
 
-General coding advices:
+- All added or changed functionality must have a test that fails before and
+  passes after changes and documents the correct behaviour.
 
--  ``from X import Y`` import form is generally preferred over
-   ``import X``
--  Absolute imports (``from library import something``) SHOULD be
-   preferred over relative imports
-   (``from .submodule import something``)
--  **is** and **is not** SHOULD be used when comparing to **None**
--  The **set** type SHOULD be used for unordered collections
--  **super()** MAY be used to call parent class methods
--  Iterators and generators SHOULD be used to iterate over large data
-   sets efficiently
--  Block comments SHOULD reference the code following them and SHOULD be
-   indented to the same level
+- The documentation (docstrings as well as ``/docs`` must be kept in sync.
 
-Tools
-~~~~~
+- New files are not allowed to skip ``pre-commit`` checks. Look for additions
+  to exclude lists in  pre-commit-config_. The only exception is that
+  test modules might skip ``mypy`` checks.
 
--  Use `Black: the uncoompromising Python code
-   formatter <https://github.com/ambv/black>`__ with not more than 120
-   characters per source line and 79 for docstrings
+- If a file currently excluded from checks in pre-commit-config_  is changed
+  substantially, the reviewer may request dropping it from the exclude list and
+  cleaning up the remaining issues.  The remaining issues must be small
+  compared to the size of the PR.
 
--  Follow NumPy format for docstrings with sphinx-Napoleon. Very useful
-   guidelines can be found in
-   `LSST <https://developer.lsst.io/python/numpydoc.html>`__ docstrings
-   conventions
+- Any comments about future work required (``#todo`` and other fixme-style
+  comments) should be either addressed before merging or deleted after an issue
+  is opened describing the future changes.
```

Comment:
```suggestion
```
We discussed this topic, ok to remove? Then let's merge.

Reply:
I agree. Let's remove this paragraph and merge the PR (or close it, it's been open for too long). @DropD Do you also agree?